frontend/src/app/features/enterprise/free-trial-button/free-trial-button.component.ts

```diff
@@ -105,20 +97,22 @@ export class FreeTrialButtonComponent implements OnInit {
       const savedDateStr = eeTrialKey.created.split(' ')[0];
       this.created = this.timezoneService.formattedDate(savedDateStr);
 
-      if (!this.eeTrialService.userData$.hasValue()) {
+      const { data } = this.eeTrialService.store.getValue();
+      if (data) {
         // after reload: get data from Augur using the trial key saved in gon
-        this.eeTrialService.trialLink = `${this.eeTrialService.baseUrlAugur}/public/v1/trials/${eeTrialKey.value}`;
-        this.getUserDataFromAugur();
+        const trialLink = `${this.eeTrialService.baseUrlAugur}/public/v1/trials/${eeTrialKey.value}`;
+        this.eeTrialService.store.update({ trialLink });
+        this.getUserDataFromAugur(trialLink);
       }
     }
   }
 
-  private getUserDataFromAugur():void {
+  private getUserDataFromAugur(trialLink:string):void {
     this.http
-      .get<TrialDetails>(`${this.eeTrialService.trialLink}/details`)
+      .get<IEnterpriseData>(`${trialLink}/details`)
       .toPromise()
-      .then((userForm:TrialDetails) => {
-        this.eeTrialService.userData$.putValue(userForm);
+      .then((data:IEnterpriseData) => {
+        this.eeTrialService.store.update({ data });
```

Comment:
this is an issue, we discussed some time ago. IMHO it is a bit weird to expose the store and work on it from here. Instead the resource service, which is understood to contain wrapping functionality, should have an `update` method and amend the store. (even if it is as trivial as here)

Reply:
Let's talk about it on Monday. No blocker for merging.

Reply:
Yes I fully agree. The trial module had to be rewritten to work with onPush. I merely translated the synchronous calls into a store so we could subscribe to its data and fix the CD issue but did nothing beyond this